frontend/apps/reader/modules/readerstatus.lua

```diff
@@ -47,27 +52,29 @@ function ReaderStatus:onEndOfBook()
 
     -- Should we start by marking the book as finished?
     if G_reader_settings:isTrue("end_document_auto_mark") then
-        self:onMarkBook(true)
+        self:markBook(true)
     end
 
-    local next_file_enabled = G_reader_settings:readSetting("collate") ~= "access"
-    local settings = G_reader_settings:readSetting("end_document_action")
+    local collate = G_reader_settings:readSetting("collate")
+    local next_file_enabled = collate ~= "access" and collate ~= "date"
+    local settings = G_reader_settings:readSetting("end_document_action") or "pop-up"
     local top_widget = UIManager:getTopmostVisibleWidget() or {}
-    if (settings == "pop-up" or settings == nil) and top_widget.name ~= "end_document" then
+    if settings == "pop-up" and top_widget.name ~= "end_document" then
         local button_dialog
         local buttons = {
             {
                 {
                     text_func = function()
-                        return self.summary.status == "complete" and _("Mark as reading") or _("Mark as finished")
+                        local status = self.ui.doc_settings:readSetting("summary").status
+                        return status == "complete" and _("Mark as reading") or _("Mark as finished")
                     end,
                     callback = function()
                         UIManager:close(button_dialog)
-                        self:onMarkBook()
+                        self:markBook()
                     end,
                 },
                 {
-                    text = _("Book status"),
+                    text = self.title,
                     callback = function()
```

Comment:
(I'm personally not a fan of factorizing text - again needing to move around to figure what it is. Moreover, translators may need some different wording whether it's in a menu or a button - or they can use a longer wording for the menu, and use a smaller one in the button.
Just writting this because here, I thought this was something new and that you were showing the _book title_ in a button.)

Reply:
As a quick reminder, if that's something we want to provide it should be something like `C_("Title", "Book status")` vs `C_("Button", "Book status")`, otherwise they'll both be the same. At the moment we're mostly taking the "wait to see if a translator complains about the lack of disambiguation" approach.